app/assets/javascripts/modules/track-selected-taxons.js

```diff
@@ -33,12 +33,12 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         var target = e.currentTarget
 
         if (target.checked) {
-          GOVUK.analytics.trackEvent('pageElementInteraction', 'checkboxClickedOn', { label: target.labels[0].innerText.trim() })
+          GOVUK.analytics.trackEvent('pageElementInteraction', 'checkboxClickedOn', { label: this.form.querySelector('label[for="' + target.id + '"]').innerText.trim() })
         } else {
-          GOVUK.analytics.trackEvent('pageElementInteraction', 'checkboxClickedOff', { label: target.labels[0].innerText.trim() })
+          GOVUK.analytics.trackEvent('pageElementInteraction', 'checkboxClickedOff', { label: this.form.querySelector('label[for="' + target.id + '"]').innerText.trim() })
```

Comment:
Opinion: it might be nicer (more DRY) to define 

```js
var label = this.form.querySelector('label[for="' + target.id + '"]').innerText.trim()
```

above, where you already have `var target`. To avoid repeating it twice in each half of this `if` statement.

Reply:
updated